src/createXcmTypes/util/getSystemChainAssetId.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+
+import { ApiPromise } from '@polkadot/api';
+
+import { SYSTEM_PARACHAINS_IDS } from '../../consts';
+import { BaseError } from '../../errors';
+import { Registry } from '../../registry';
+import { foreignAssetMultiLocationIsInRegistry } from './foreignAssetMultiLocationIsInRegistry';
+import { getChainIdBySpecName } from './getChainIdBySpecName';
+import { isValidForeignAssetMultiLocation } from './isValidForeignAssetMultiLocation';
+/**
+ * Returns the correct asset id for a valid system chain token symbol
+ * integer id or foreign asset multilocation
+ * errors if given an invalid symbol, integer id or multilocation that is
+ * not registered or found in the chains state
+ *
+ * @param tokenSymbol string
+ * @param specName string
+ */
+export const getSystemChainAssetId = async (
```

Comment:
What do we think about making the `_api` the first arg here. I think it would be nice if we try and make the api come first wherever applicable as it will help with consistency.

Reply:
makes sense! I've updated all similar cases to have `api` as the first arg